plugins/speccy.py

```diff
@@ -0,0 +1,123 @@
+# -*- coding: utf-8 -*-
+
+from cloudbot import hook
+
+import re
+import requests
+
+from bs4 import BeautifulSoup as BS
+from contextlib import closing
+
+url_re = re.compile('https?:\/\/speccy\.piriform\.com\/results\/[a-zA-Z0-9]+', re.I)
+
+GPU_RE = re.compile('.*(amd|radeon|intel|integrated|nvidia|geforce|gtx).*\n.*', re.I)
+PICO_RE = re.compile('.*pico', re.I)
+KMS_RE = re.compile('.*kms', re.I)
+BOOSTER_RE = re.compile('.*booster', re.I)
+REVIVER_RE = re.compile('.*reviver', re.I)
+KILLER_RE = re.compile('.*killer.+service', re.I)
+
+
+@hook.regex(url_re)
+def get_speccy_url(match):
+
+    with closing(requests.get(match.group(), )) as response:
+        soup = BS(response.content, "lxml", from_encoding=response.encoding)
+
+    body = soup.body
+
+    data = []
+
+    def specout():
+        spec_out = re.sub(r"\s+", " ", ' ● '.join(data))
+        return spec_out
+
+    os_spec = body.find("div", text='Operating System')
+    if os_spec:
+        data.append(
+            "\x02OS:\x02" + " " + os_spec.next_sibling.next_sibling.text)
+
+    def cspace():
+        c_drive = body.find("div", class_="datavalue", text='C:')
+        if c_drive is not None:
+            c_drive = body.parent.parent
+            c_free_space_data = c_drive.find(
+                "div", text='Free Space:\xA0', class_="datakey").parent
+            return c_free_space_data.find(class_="datavalue").text
+
+    if cspace():
+        data.append("\x02Space Left (C:\):\x02" + " " + cspace())
+
+    ram_spec = body.find("div", text='RAM')
+    if ram_spec:
+        data.append(
+            "\x02RAM:\x02" + " " + ram_spec.next_sibling.next_sibling.text)
+
+    ram_usg_spec = body.find(
+        "div", class_="blue clear",
+        text='Physical Memory').next_sibling.next_sibling.find(
+            "div", text='Memory Usage:\xA0',
+            class_="datakey").parent.find(class_="datavalue").text
```

Comment:
if `body.find("div", class_="blue clear", text='Physical Memory')` returns `None` here, it will cause an error. You should do a `None` check like you do in the rest of the function.